presto-native-execution/presto_cpp/main/SystemConnector.cpp

```diff
@@ -0,0 +1,174 @@
+/*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+#include "presto_cpp/main/SystemConnector.h"
+#include "presto_cpp/main/PrestoTask.h"
+#include "presto_cpp/main/TaskManager.h"
+
+namespace facebook::presto {
+
+using namespace velox;
+
+SystemTableHandle::SystemTableHandle(std::string connectorId, std::string schemaName, std::string tableName) :
+      ConnectorTableHandle(std::move(connectorId)), schemaName_(std::move(schemaName)), tableName_(std::move(tableName)){
+  VELOX_USER_CHECK_EQ(schemaName_, "runtime", "SystemConnector supports only runtime schema");
+  VELOX_USER_CHECK_EQ(tableName_, "tasks", "SystemConnector supports only tasks table");
+
+  std::vector<std::string> kTaskColumnNames =
+      {"node_id", "task_id", "stage_execution_id", "stage_id", "query_id", "state", "splits", "queued_splits", "running_splits", "completed_splits",
+       "split_scheduled_time_ms", "split_cpu_time_ms", "split_blocked_time_ms", "raw_input_bytes", "raw_input_rows", "processed_input_bytes",
+       "processed_input_rows", "output_bytes", "output_rows", "physical_written_bytes", "created", "start", "last_heartbeat", "end"};
+  std::vector<velox::TypePtr> kTaskColumnTypes =
```

Comment:
const

Reply:
RowTypePtr requires non-const column names and types list. So this cannot be changed.